dot/state/pruner.go

```diff
@@ -0,0 +1,363 @@
+package state
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+)
+
+const (
+	journalPrefix = "journal"
+	lastPrunedKey = "last_pruned"
+	pruneInterval = time.Second
+)
+
+// nolint
+const (
+	FullNode    = "full"
+	ArchiveNode = "archive"
+)
+
+// Pruning online pruning mode of historical state tries
+type Pruning string
```

Comment:
```suggestion
type PruningMode string
```
consider moving this into a subpackage called `pruning`.  That way this can be referenced as `pruning.Mode`

Reply:
Done. Moved the code a pruner package.